src/models/generated/GENROU.jl

```diff
@@ -0,0 +1,185 @@
+        Td0_pp::Float64
+        Tq0_p::Float64
+        Tq0_pp::Float64
+        Xd::Float64
+        Xq::Float64
+        Xd_p::Float64
+        Xq_p::Float64
+        Xd_pp::Float64
+        Xl::Float64
+        S1_0::Float64
+        S1_2::Float64
+        ext::Dict{String, Any}
+        states::Vector{Symbol}
+        n_states::Int64
+        internal::InfrastructureSystemsInternal
+    end
+
+Parameters of 4-states round-rotor synchronous machine: GENROU model
+
+# Arguments
+- `R::Float64`: Armature resistance, validation range: (0, nothing)
+- `Td0_p::Float64`: Time constant of transient d-axis voltage, validation range: (0, nothing)
+- `Td0_pp::Float64`: Time constant of sub-transient d-axis voltage, validation range: (0, nothing)
+- `Tq0_p::Float64`: Time constant of transient q-axis voltage, validation range: (0, nothing)
+- `Tq0_pp::Float64`: Time constant of sub-transient q-axis voltage, validation range: (0, nothing)
+- `Xd::Float64`: Reactance after EMF in d-axis per unit, validation range: (0, nothing)
+- `Xq::Float64`: Reactance after EMF in q-axis per unit, validation range: (0, nothing)
+- `Xd_p::Float64`: Transient reactance after EMF in d-axis per unit, validation range: (0, nothing)
+- `Xq_p::Float64`: Transient reactance after EMF in q-axis per unit, validation range: (0, nothing)
+- `Xd_pp::Float64`: Sub-Transient reactance after EMF in d-axis per unit. Note: Xd_pp = Xq_pp, validation range: (0, nothing)
+- `Xl::Float64`: Stator leakage reactance, validation range: (0, nothing)
+- `S1_0::Float64`: Saturation factor at 1 pu flux, validation range: (0, nothing)
+- `S1_2::Float64`: Saturation factor at 1 pu flux, validation range: (0, nothing)
+- `ext::Dict{String, Any}`
+- `states::Vector{Symbol}`: The states are:
+	eq_p: q-axis generator voltage behind the transient reactance,
+	ed_p: d-axis generator voltage behind the transient reactance,
+	ψ_kd: flux linkage in the first equivalent damping circuit in the d-axis,
+	ψ_kq: flux linkage in the first equivalent damping circuit in the d-axis
+- `n_states::Int64`
+- `internal::InfrastructureSystemsInternal`: power system internal reference, do not modify
+"""
+mutable struct GENROU <: Machine
+    "Armature resistance"
+    R::Float64
+    "Time constant of transient d-axis voltage"
+    Td0_p::Float64
+    "Time constant of sub-transient d-axis voltage"
+    Td0_pp::Float64
+    "Time constant of transient q-axis voltage"
+    Tq0_p::Float64
+    "Time constant of sub-transient q-axis voltage"
+    Tq0_pp::Float64
+    "Reactance after EMF in d-axis per unit"
+    Xd::Float64
+    "Reactance after EMF in q-axis per unit"
+    Xq::Float64
+    "Transient reactance after EMF in d-axis per unit"
+    Xd_p::Float64
+    "Transient reactance after EMF in q-axis per unit"
+    Xq_p::Float64
+    "Sub-Transient reactance after EMF in d-axis per unit. Note: Xd_pp = Xq_pp"
+    Xd_pp::Float64
+    "Stator leakage reactance"
+    Xl::Float64
+    "Saturation factor at 1 pu flux"
```

Comment:
We need to document better the S1 and s0 factors

Reply:
See changes in 78eca89

Reply:
We should store S in a tuple and saturation_coefficients (A,B) also in a tuple. We should include in supplemental_constructors how to obtain those coefficients.